tests/controllers_tests/controllers_test.go

```diff
@@ -243,27 +257,27 @@ func TestGenerateCSV(t *testing.T) {
 	}
 	conceptIds := []int64{10, 22}
 
-	b := controllers.GenerateCSV(
+	csvLines := controllers.GeneratePartialCSV(
```

Comment:
I think we need separate tests, one for each of these methods. Please keep the original test intact and make a new one for your new method. The issue is that now the `GenerateCSV` has no tests of its own anymore, and I think its original test was a good test.

Reply:
i think i'm a bit confused here, i changed the original `GenerateCSV` method into `GeneratePartialCSV`. The difference is i'm returning the csv rows instead of writing them to a csv and returning the file pointer. So this test is the same as original but instead of checking csv it checks the rows that would generate the csv.

Let me know if this makes sense

Reply:
Ah, I see. Makes sense. Then it would be good to have the "A value with, comma!" move to the `TestGenerateCompleteCSV`. The goal of this simple test was to ensure that commas in CSV strings are correct "escaped" by having their sentence value quoted.

Reply:
ah makes sense, let me add that